SpatialGDK/Source/SpatialGDK/Private/Interop/Connection/SpatialEventTracer.cpp

```diff
@@ -74,37 +96,54 @@ SpatialEventTracer::SpatialEventTracer(const FString& WorkerId)
+	UE_LOG(LogSpatialEventTracer, Log, TEXT("Spatial event tracing enabled."));
 
 	UEventTracingSamplingSettings* SamplingSettings = Settings->GetEventTracingSamplingSettings();
 
-	UE_LOG(LogSpatialEventTracer, Log, TEXT("Setting event tracing sampling probability. Probability: %f."),
-		   SamplingSettings->SamplingProbability);
-
 	TArray<Trace_SpanSamplingProbability> SpanSamplingProbabilities;
-	TArray<std::string> AnsiStrings; // Worker requires ansi const char*
-	for (const auto& Pair : SamplingSettings->EventSamplingModeOverrides)
+	FSpatialTraceEventDataBuilder::FStringCache
+		AnsiStrings; // Storage for strings passed to the worker SDK Worker requires ansi const char*
+
+	if (Settings->bCaptureAllEventTracingData)
 	{
-		const FString& EventName = Pair.Key.ToString();
-		UE_LOG(LogSpatialEventTracer, Log, TEXT("Adding trace event sampling override. Event: %s Probability: %f."), *EventName,
-			   Pair.Value);
-		int32 Index = AnsiStrings.Add(TCHAR_TO_ANSI(*EventName));
-		SpanSamplingProbabilities.Add({ AnsiStrings[Index].c_str(), Pair.Value });
+		UE_LOG(LogSpatialEventTracer, Log, TEXT("Setting event tracing span sampling always."));
+		Parameters.span_sampling_parameters.sampling_mode = Trace_SamplingMode::TRACE_SAMPLING_MODE_ALWAYS;
+	}
+	else
+	{
+		Parameters.span_sampling_parameters.sampling_mode = Trace_SamplingMode::TRACE_SAMPLING_MODE_PROBABILISTIC;
+
+		UE_LOG(LogSpatialEventTracer, Log, TEXT("Setting event tracing span sampling probabalistic. Probability: %f."),
+			   SamplingSettings->SamplingProbability);
+
+		for (const auto& Pair : SamplingSettings->EventSamplingModeOverrides)
+		{
+			const FString& EventName = Pair.Key.ToString();
+			UE_LOG(LogSpatialEventTracer, Log, TEXT("Adding trace event sampling override. Event: %s Probability: %f."), *EventName,
+				   Pair.Value);
+			SpanSamplingProbabilities.Add({ AnsiStrings.Get(AnsiStrings.AddFString(EventName)), Pair.Value });
+		}
+
+		Parameters.span_sampling_parameters.probabilistic_parameters.default_probability = SamplingSettings->SamplingProbability;
+		Parameters.span_sampling_parameters.probabilistic_parameters.probability_count = SpanSamplingProbabilities.Num();
+		Parameters.span_sampling_parameters.probabilistic_parameters.probabilities = SpanSamplingProbabilities.GetData();
 	}
 
-	SamplingParameters.probabilistic_parameters.default_probability = SamplingSettings->SamplingProbability;
-	SamplingParameters.probabilistic_parameters.probability_count = SpanSamplingProbabilities.Num();
-	SamplingParameters.probabilistic_parameters.probabilities = SpanSamplingProbabilities.GetData();
+	// Filters
+	TraceQueryPtr PreFilter = ParseOrDefault(SamplingSettings->EventPreFilter, "pre-filter");
+	TraceQueryPtr PostFilter = ParseOrDefault(SamplingSettings->EventPostFilter, "post-filter");
 
-	Trace_EventTracer_SetSampler(EventTracer, &SamplingParameters);
+	checkf(PreFilter.Get() != nullptr, TEXT("Pre-filter is invalid."));
```

Comment:
Hmm, should these be checks? I feel like it is quite easy for a user to enter an invalid query?

Reply:
I guess your post property change code goes someway ro prevent this

Reply:
Yeah, ParseOrDefault will also fallback to "false" if the queries are invalid so it should always be set (unless something went horribly wrong)

Reply:
Ahh yes that is also true! Nice!